lms/djangoapps/grades/models.py

```diff
@@ -181,37 +155,41 @@ def blocks(self):
         """
         return BlockRecordList.from_json(self.blocks_json)
 
-
-class PersistentSubsectionGradeQuerySet(models.QuerySet):
-    """
-    A custom QuerySet, that handles creating a VisibleBlocks model on creation, and
-    extracts the course id from the provided usage_key.
-    """
-    def create(self, **kwargs):
+    @classmethod
+    def bulk_read(cls, course_key):
         """
-        Instantiates a new model instance after creating a VisibleBlocks instance.
+        Reads all visible block records for the given course.
 
         Arguments:
-            user_id (int)
-            usage_key (serialized UsageKey)
-            course_version (str)
-            subtree_edited_timestamp (datetime)
-            earned_all (float)
-            possible_all (float)
-            earned_graded (float)
-            possible_graded (float)
-            visible_blocks (iterable of BlockRecord)
-        """
-        visible_blocks = kwargs.pop('visible_blocks')
-        kwargs['course_version'] = kwargs.get('course_version', None) or ""
-        if not kwargs.get('course_id', None):
-            kwargs['course_id'] = kwargs['usage_key'].course_key
-
-        visible_blocks_hash = VisibleBlocks.objects.hash_from_blockrecords(BlockRecordList.from_list(visible_blocks))
-        return super(PersistentSubsectionGradeQuerySet, self).create(
-            visible_blocks_id=visible_blocks_hash,
-            **kwargs
-        )
+            course_key: The course identifier for the desired records
+        """
+        return cls.objects.filter(course_id=course_key)
+
+    @classmethod
+    def bulk_create(cls, block_record_lists, course_key):
+        """
+        Bulk creates VisibleBlocks for the given iterator of
+        BlockRecordList objects for the given course_key.
+        """
+        return cls.objects.bulk_create([
+            VisibleBlocks(
+                blocks_json=brl.json_value,
+                hashed=brl.hash_value,
+                course_id=course_key,
+            )
+            for brl in block_record_lists
+        ])
```

Comment:
BlockRecordList now contains a course key.  Do we want to validate that we're providing BRLs that match the course_key provided?